### ISSUE TITLE
Support for Iceberg FOR SYSTEM_VERSION AS OF

### ISSUE THREAD

**Issue by jasonamyers** [laughingman7743/PyAthena#517]
When used with Iceberg tables, Athena can restrict the table by version ID and timestamp. I'd love to be able to leverage this in my SQLAlchemy queries. More details can be found at https://docs.aws.amazon.com/athena/latest/ug/querying-iceberg-table-data.html.

**Comment by laughingman7743** [comment]
What use cases do you have in mind?
Do you want to construct the time travel specification functionally when assembling the query? Or do you want to be able to specify timestamps and versions in the model definition?

I am not that familiar with extending SQLAlchemy. If you have any SQLAlchemy implementations in other databases that might be helpful, please let me know.

**Comment by jasonamyers** [comment]
I'd love to either do it during the table's metadata build-up, specify a version, and have it reflected in the resulting default_from clauses or make a manual call to set the version for each table if desired.

**Comment by jasonamyers** [comment]
I'm asking in SQLAlchemy also if they have thoughts on this https://github.com/sqlalchemy/sqlalchemy/discussions/11161.

**Comment by jasonamyers** [comment]
I added a super naive function to test. This leverages the SQLAlchemy `with_hints` functionality.


*Compiler Support:*
```python
class AthenaStatementCompiler(SQLCompiler):

    def get_from_hint_text(self, table, text):
        return text
```

*Usage Example:*
```python
engine = get_engine('athena')
table = build_table(engine, 'my_table', 'my_schema')
ta = table.alias()
version = 8860342768238431084
query = select(func.count(ta.c.record_id)).with_hint(ta, f'FOR VERSION AS OF {version}')
result = str(query.compile(compile_kwargs={"literal_binds": True}, dialect=AthenaDialect()))
```

*Output:*
```shell
SELECT count(my_table.record_id) AS count_1 
FROM data_lake_curated.my_table AS my_table_1 FOR VERSION AS OF 8860342768238431084
```

**Comment by jasonamyers** [comment]
@laughingman7743 any chance we can get a release with this is in? I wanna move our usages to it.

**Comment by laughingman7743** [comment]
I just released v3.5.0. 🎉 
https://github.com/laughingman7743/PyAthena/releases/tag/v3.5.0
https://pypi.org/project/PyAthena/

**Comment by jasonamyers** [comment]
You rock thank you
